File: Serveur/Rectangle.java
public class Rectangle extends Formes{
    private int hauteur;
    private int largeur;

    public Rectangle(Point p, int hauteur, int largeur) throws Erreur{
        this.setP(p);
        this.setHauteur(hauteur);
        this.setLargeur(largeur);
    }

    public int getHauteur() {
        return hauteur;
    }

    public void setHauteur(int hauteur) throws Erreur{
        if(hauteur < 0)
            throw new Erreur("Hauteur inferieur à 0");
        this.hauteur = hauteur;
    }

    public int getLargeur() {
        return largeur;
    }

    public void setLargeur(int largeur) throws Erreur{
        if(largeur < 0)
            throw new Erreur("Largeur inferieur à 0");
        this.largeur = largeur;
    }
}

File: Client/Rectangle.cpp
//
// Created by Samvel on 02/02/2022.
//

#include "Rectangle.h"

Rectangle::Rectangle(const char *s, double hauteur, double largeur) : Forme(s), hauteur(hauteur), largeur(largeur) {}

double Rectangle::getHauteur() const {
    return hauteur;
}

void Rectangle::setHauteur(double hauteur) {
    this->hauteur = hauteur;
}

double Rectangle::getLargeur() const {
    return largeur;
}

void Rectangle::setLargeur(double largeur) {
    this->largeur = largeur;
}

string Rectangle::toString() const {
    return "Rectangle " + getVecteur().toString() + " : hauteur=" + to_string(getHauteur()) + ", largeur=" + to_string(getLargeur());
}

ostream &operator<<(ostream &os, const Rectangle &rectangle) {
    return os << rectangle.toString();
}

void Rectangle::translation(Vecteur2D v) {
    setVecteur(getVecteur() + v);
}
